src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,417 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+import AbstractAlgebra: Field
+
+export
+    iscongruent
+
+###############################################################################################################
+
+# Algorithm
+
+###############################################################################################################
+
+# The functions _find_radical, _block_anisotropic_elim and _block_herm_elim are based on the article
+# James B. Wilson, Optimal Algorithms of Gram-Schmidt type, Linear Algebra and its Applications 438 (2013) 4573-4583
+
+# returns x,y such that ax^2+by^2 = c
+# at the moment, it simply search for x such that (c-ax^2)/b is a square
+# TODO: is there a faster way to find x and y?
+# TODO: it would be better if this is deterministic. This depends on gen(F) and issquare(F).
+function _solve_eqn(a::T, b::T, c::T) where T <: FinFieldElem
+   F = parent(a)  
+   for x in F
+      s = (c - a*x^2)*b^-1
+#      vero, y = issquare(s)
+      if issquare(s) return x,square_root(s) end
+   end
+   return nothing, nothing
+end
+
+
+
+# if _is_symmetric, returns C,A,d where A*B*transpose(frobenius(A,e)) = C, C = block_matrix(2,2,[C,0,0,0]) and d = rank(C)
+# else returns C,A,d where B*A = C, C = [C 0] and d = rank(C)
+# Assumption: if _is_symmetric==true, then nr=nc always
+# Assumption: e = deg(F)/2 in the hermitian case, e=0 otherwise
+function _find_radical(B::MatElem{T}, F::Field, nr::Int, nc::Int; e::Int=0, _is_symmetric::Bool=false) where T <: FinFieldElem
+
+   @assert !_is_symmetric || nr==nc
+   V1 = VectorSpace(F,nc)
+   V2 = VectorSpace(F,nr)
+   K, embK = kernel(ModuleHomomorphism(V1, V2, _is_symmetric ? B : transpose(B)))
+   U, embU = complement(V1,K)
+   d = dim(U)
+   type_vector = elem_type(V1)
+   A = matrix(vcat(type_vector[embU(v) for v in gens(U)], type_vector[embK(v) for v in gens(K)] ))
```

Comment:
OK, here's another attempt that deals with trivial subspaces
```suggestion
   elemT = elem_type(V1)
   A = matrix(elemT[embU.(gens(U)) ; embK.(gens(K))])
```